src/common/task_system/task_scheduler.cpp

```diff
@@ -144,5 +151,15 @@ void TaskScheduler::runWorkerThread() {
     }
 }
 
+void TaskScheduler::interruptTaskIfTimeOutNoLock(
+    processor::ExecutionContext* context, uint64_t scheduledTaskID) {
+    if (context->clientContext->isQueryTimeOut()) {
+        context->clientContext->interrupt();
+        removeErroringTask(scheduledTaskID);
```

Comment:
If you interrupt, all threads executing current task will be notified through the atomic flag and throwing exceptions. There is no need to remove erroring task because at l71 you will be waiting for the task to complete. Then l76 you will find the task has exception (the timeout exception. Although we have the bug that only one exception can be caught. We guarantee to cache here). And l77 you will remove erroring task.

Reply:
removed the call to removeErroringTask